drivers/hmis/spec/system/hmis/mci_clearance_spec.rb

```diff
@@ -0,0 +1,262 @@
+    before(:each) do
+      visit "/client/#{c2.id}/profile/edit"
+    end
+
+    context 'with auto-clearance result for client that already exists in HMIS' do
+      before(:each) do
+        result = [
+          HmisExternalApis::AcHmis::MciClearanceResult.new(
+            mci_id: c1_mci_id.value,
+            score: 99, # above auto-clearance threshold (97)
+            client: build(:hmis_hud_client, first_name: 'quentin', man: 1),
+            existing_client_id: c1.id,
+          ),
+        ]
+        allow(stub_mci).to receive(:clearance).and_return(result)
+      end
+
+      it 'allows linking to MCI ID even though its a duplicate client' do
+        click_button 'Search for MCI ID'
+        assert_text 'Client already exists in HMIS'
+        assert has_field?("select_mci_#{c1_mci_id.value}", visible: :all, disabled: false) # dup is selectable
+        assert_no_text 'New MCI ID' # not allowed to create a new MCI ID, because they auto-cleared
+
+        # Choose the duplicate MCI ID (allow because editing, normally not allowed)
+        find_by_id("select_mci_#{c1_mci_id.value}", visible: :all).set(true)
+        expect do
+          click_button 'Save Changes'
+        end.to change(HmisExternalApis::ExternalId.mci_ids, :count).by(1)
+        expect(c2.reload.ac_hmis_mci_ids.count).to eq(1)
+        expect(c2.reload.ac_hmis_mci_ids.first.value).to eq(c1_mci_id.value)
+      end
+    end
+  end
+
+  context 'Creating a new client for enrollment in PH project' do
+    before(:each) do
+      visit "/projects/#{p1.id}/add-household"
+      fill_in 'Search Clients', with: 'xxx'
+      click_button 'Search'
+      click_button 'Add New Client'
+      assert_text 'Enroll a New Client'
+
+      mui_select unit.name, from: 'Unit'
+      enter_client_details
+    end
+
+    context 'with no clearance results' do
+      before(:each) do
+        allow(stub_mci).to receive(:clearance).and_return([])
+      end
+
+      it 'creates a new MCI ID by default' do
+        click_button 'Search for MCI ID'
+        assert_text 'No Matches Found'
+        assert_text 'New MCI ID'
+        assert has_checked_field?('select_create_new_mci', disabled: true, visible: :all) # checked by default
+
+        expect do
+          click_button 'Create & Enroll Client'
+        end.to change(Hmis::Hud::Client, :count).by(1)
+
+        # Not mocking the actual ExternalId creation here, just ensuring that it was invoked
+        expect(stub_mci).to have_received(:create_mci_id)
+      end
+    end
+
+    context 'with auto-clearance result for client that already exists in HMIS' do
+      before(:each) do
+        result = [
+          HmisExternalApis::AcHmis::MciClearanceResult.new(
+            mci_id: c1_mci_id.value,
+            score: 99, # above auto-clearance threshold (97)
+            client: build(:hmis_hud_client, first_name: 'quentin', man: 1),
+            existing_client_id: c1.id,
+          ),
```

Comment:
Maybe not worth the bother but this could be DRYed up with a factory:
```ruby
# not tested
FactoryBot.define do
  factory :mci_clearance_result, class: 'HmisExternalApis::AcHmis::MciClearanceResult' do
    mci_id { SecureRandom.uuid }
    score { 99 }
    client { association :hmis_hud_client, first_name: 'quentin', man: 1 }
    # existing_client_id { create(:hmis_hud_client).id }

    initialize_with do
      new(attributes)
    end
  end
end

###
create(:mci_clearance_result, mci_id: c1_mci_id.value,  existing_client_id: c1.id)

```